framework/include/eventloop/eventloop.h

```diff
@@ -0,0 +1,99 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/**
+ * @defgroup Eventloop Eventloop
+ * @ingroup Eventloop
+ * @brief Provides APIs for Eventloop
+ * @{
+ */
+/**
+ * @file eventloop/eventloop.h
+ */
+#ifndef __EVENTLOOP_H__
+#define __EVENTLOOP_H__
+
+#include <libtuv/uv.h>
+#include <stdbool.h>
+
+/**
+ * @brief Error Type of Result Value returned from Event Loop
+ * @details If operation is failed, these defined values will be returned.
+ */
+enum el_result_error_e {
+	EVENTLOOP_INVALID_PARAM = -1,
+	EVENTLOOP_BUSY = -2,
+	EVENTLOOP_OUT_OF_MEMORY = -3,
+};
+
+/**
+ * @brief EventLoop Timer structure
+ */
+typedef uv_timer_t el_timer_t;
+
+/**
+ * @brief EventLoop Timeout Callback
+ */
+typedef void (*timeout_callback)(void *);
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/**
+ * @brief Create timer to call the function in the given period of time
+ * @details @b #include <eventloop/eventloop.h>
+ * @param[in] timeout interval in milliseconds for calling callback function repeatly
```

Comment:
Could you fill more details?
1. timeout from adding timer or starting loop?
2. repeatly is not always right. Let's remove this word from this line. Below line is enough.